fw/pool.c

```diff
@@ -152,8 +151,14 @@ __tfw_pool_alloc(TfwPool *p, size_t n, bool align, bool *new_page)
 			: (void *)(c + 1);
 	}
 
-	a = (char *)TFW_POOL_CHUNK_END(p);
-	p->off += n;
+	if (align) {
+		a = (void *)TFW_POOL_ALIGN_SZ((unsigned long)TFW_POOL_CHUNK_END(p));
+		p->off = TFW_POOL_ALIGN_SZ(p->off) + n;
+	}
+	else {
+		a = (void *)TFW_POOL_CHUNK_END(p);
+		p->off += n;
+	}
```

Comment:
There are 2 tests for `align` value on the fast path, which are just one after another, but the compiler is smart enough to make only one test. Also the compiler is smart enough to optimize the second assignment to `p->off = p_off + n`. Please write a comment, that we actually don't have 2 conditional jumps here and don't spend anything on double offset computation.

But I think the code would be still simpler with:
```C
p_off = (likely(align) ? TFW_POOL_ALIGN_SZ(p->off) : p->off) + n;
if (unlikely(p_off > TFW_POOL_CHUNK_SZ(p))) {
   // ....
}
if (likely(align))
    a = (void *)TFW_POOL_ALIGN_SZ((unsigned long)TFW_POOL_CHUNK_END(p));
else
   a = (void *)TFW_POOL_CHUNK_END(p);
p->off = p_off;
```
Also it'd be better to use `likely()` for align as the most code uses the aligned version